### ISSUE TITLE
Sticker Thumbnail Preview Transparency Issue

### ISSUE THREAD

**Issue by AverytheFurry** [matrix-org/dendrite#1815]
### Background information

- **Dendrite version or git SHA**: dendrite 0.3.11-1(https://aur.archlinux.org/packages/dendrite/)
- **Monolith or Polylith?**: Monolith
- **SQLite3 or Postgres?**: Postgres
- **Running in Docker?**: No
- **`go version`**: go 2:1.16.2-1

### Description

Stickers sometimes have a non-transparent background (using https://github.com/maunium/stickerpicker)
I asked the stickerpicker dev about it and they told me it was most likely dendrite.

### Steps to reproduce

- Create a stickerpicker instance and implement in element
- Send stickers, most of them will have a black background, but some will have a transparent background

<!--
Describe how what happens differs from what you expected.

If you can identify any relevant log snippets from server logs, please include
those (please be careful to remove any personal or private data). Please surround them with
``` (three backticks, on a line on their own), so that they are formatted legibly.

Alternatively, please send logs to @kegan:matrix.org or @neilalexander:matrix.org
with a link to the respective Github issue, thanks!
-->
What is expected is that the background would be transparent.
![2021-04-03_03-18](https://user-images.githubusercontent.com/47131867/113475639-74146200-9466-11eb-8fcf-b655802744c9.png)

**Comment by ShadowJonathan** [comment]
(Maybe it would help to add the MXC URLs to those three stickers there, to be able to inspect them)

**Comment by AverytheFurry** [comment]
> (Maybe it would help to add the MXC URLs to those three stickers there, to be able to inspect them)

MXC? Do you mean the link to the image file?

https://matrix-client.matrix.org/_matrix/media/r0/thumbnail/matrix.org/XBvLmfZMDqSnlnTzwBbhaGic?height=128&width=128&method=scale
https://matrix-client.matrix.org/_matrix/media/r0/thumbnail/matrix.org/whhVHLOuOALIArEMLxmtsbib?height=128&width=128&method=scale
https://matrix-client.matrix.org/_matrix/media/r0/thumbnail/matrix.org/jQGjxWzTIfhqVBpPBQKjOLcd?height=128&width=128&method=scale

**Comment by ShadowJonathan** [comment]
Yes, i'll thumbnail these via `dendrite.matrix.org`, since it seems to have the same issue. Curiously enough, these images also dont scale properly (as is instructed via the `query` parameter)

<details><summary>DMO Images</summary>

`mxc://matrix.org/XBvLmfZMDqSnlnTzwBbhaGic`
![](https://dendrite.matrix.org/_matrix/media/r0/thumbnail/matrix.org/XBvLmfZMDqSnlnTzwBbhaGic?height=128&width=128&method=scale)

`mxc://matrix.org/jQGjxWzTIfhqVBpPBQKjOLcd`
![](https://dendrite.matrix.org/_matrix/media/r0/thumbnail/matrix.org/jQGjxWzTIfhqVBpPBQKjOLcd?height=128&width=128&method=scale)

`mxc://matrix.org/whhVHLOuOALIArEMLxmtsbib`
![](https://dendrite.matrix.org/_matrix/media/r0/thumbnail/matrix.org/whhVHLOuOALIArEMLxmtsbib?height=128&width=128&method=scale)

`mxc://matrix.org/MuqbQlrpCcbBCXvEstjzpwVv` (the "growl" one from above)
![](https://dendrite.matrix.org/_matrix/media/r0/thumbnail/matrix.org/MuqbQlrpCcbBCXvEstjzpwVv?height=128&width=128&method=scale)

</details>

`identify` and `file` dont find anything weird with these.

`matrix.org` (a synapse server) seems to find no issue with these:

<details><summary>MO Images</summary>

`mxc://matrix.org/XBvLmfZMDqSnlnTzwBbhaGic`
![](https://matrix.org/_matrix/media/r0/thumbnail/matrix.org/XBvLmfZMDqSnlnTzwBbhaGic?height=128&width=128&method=scale)

`mxc://matrix.org/jQGjxWzTIfhqVBpPBQKjOLcd`
![](https://matrix.org/_matrix/media/r0/thumbnail/matrix.org/jQGjxWzTIfhqVBpPBQKjOLcd?height=128&width=128&method=scale)

`mxc://matrix.org/whhVHLOuOALIArEMLxmtsbib`
![](https://matrix.org/_matrix/media/r0/thumbnail/matrix.org/whhVHLOuOALIArEMLxmtsbib?height=128&width=128&method=scale)

`mxc://matrix.org/MuqbQlrpCcbBCXvEstjzpwVv` (the "growl" one from above)
![](https://matrix.org/_matrix/media/r0/thumbnail/matrix.org/MuqbQlrpCcbBCXvEstjzpwVv?height=128&width=128&method=scale)

</details>

**Comment by AverytheFurry** [comment]
I'm planning on changing the webserver I host these on, will that cause any issues?

**Comment by AverytheFurry** [comment]
> Yes, i'll thumbnail these via `dendrite.matrix.org`, since it seems to have the same issue. Curiously enough, these images also dont scale properly (as is instructed via the `query` parameter)
> DMO Images
> 
> `identify` and `file` dont find anything weird with these.
> 
> `jboi.nl` (a synapse server, my own) seems to find no issue with these:
> Synapse Images
> 
> `matrix.org` seems to have these wrong cached, by the way, giving a larger image than needed.

also those synapse images appear to be very small, I like the current size of the stickers (they aren't emojis)

**Comment by ShadowJonathan** [comment]
Synapse is displaying the thumbnail sizes correctly, scaling it to 128, its matrix.org that has this cached incorrectly.

> I'm planning on changing the webserver I host these on, will that cause any issues with the images you posted?

I'm not referencing your server domain for privacy concerns, but unless you also change the domain (which has its own can of worms), everything should be alright.

**Comment by AverytheFurry** [comment]
> Synapse is displaying the thumbnail sizes correctly, scaling it to 128, its matrix.org that has this cached incorrectly.
> 
> > I'm planning on changing the webserver I host these on, will that cause any issues with the images you posted?
> 
> I'm not referencing your server domain for privacy concerns, but unless you also change the domain (which has its own can of worms), everything should be alright.

I am going to change the domain lol

**Comment by ShadowJonathan** [comment]
> I am going to change the domain lol (where the stickers are hosted)

That's not going to fix the problem, to dendrite, they'd be the same files, regardless if they're hosted on other servers, or the same one.

**Comment by AverytheFurry** [comment]
> > I am going to change the domain lol (where the stickers are hosted)
> 
> That's not going to fix the problem, to dendrite, they'd be the same files, regardless if they're hosted on other servers, or the same one.

Not trying to fix the issue lol

**Comment by Kegsay** [comment]
This may be a thumbnailing issue in Dendrite, needs investigation.

**Comment by yukiqt** [comment]
Having the same issue.

**Comment by fabrixxm** [comment]
I would say the issue is coming from [here](https://github.com/matrix-org/dendrite/blob/db83789654ade3cf4f900e8fbcaa742b60c5dc6c/mediaapi/thumbnailer/thumbnailer_nfnt.go#L214) and [here](https://github.com/matrix-org/dendrite/blob/db83789654ade3cf4f900e8fbcaa742b60c5dc6c/mediaapi/thumbnailer/thumbnailer_bimg.go#L176)
